src/Lachain.Consensus/Messages/MessageEnvelopeRepositoryManager.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using Lachain.Logger;
+using Lachain.Storage.Repositories;
+
+namespace Lachain.Consensus.Messages
+{
+    public class MessageEnvelopeRepositoryManager
+    {
+        private readonly IMessageEnvelopeRepository _repository;
+        private MessageEnvelopeList? _messageEnvelopeList;
+        private static readonly ILogger<MessageEnvelopeRepositoryManager> logger = LoggerFactory.GetLoggerForClass<MessageEnvelopeRepositoryManager>();
+
+        public bool IsPresent => !(_messageEnvelopeList is null);
+        public MessageEnvelopeRepositoryManager(IMessageEnvelopeRepository repository)
+        {
+            _repository = repository;
+            var bytes = repository.LoadMessages();
```

Comment:
Instead of loading from db in constructor, maybe we can load from db only if we need restoration. This can be considered as an optimization I think.